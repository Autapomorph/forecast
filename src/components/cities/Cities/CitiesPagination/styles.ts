import styled, { css } from 'styled-components/macro';

type Props = {
  isSinglePage: boolean;
};

export const CitiesPagination = styled.div`
  .pagination {
    display: flex;
    justify-content: flex-end;
    list-style: none;

    ${({ isSinglePage }: Props) =>
      isSinglePage &&
      css`
        display: none;
      `}
  }

  .page {
    background: var(--white);
    color: var(--contrast-text-color);
    transition: 0.3s;
    cursor: pointer;

    :hover {
      background: var(--gray);
    }

    &--prev {
      border-radius: 5px 0 0 5px;
    }

    &--next {
      border-radius: 0 5px 5px 0;
    }

    &--disabled {
      background: var(--light-gray);
      color: var(--disabled-text-color);
      cursor: default;

      :hover {
        background: var(--light-gray);
        color: var(--disabled-text-color);
      }
    }

    &--hidden {
      display: none;
    }

    a {
      display: block;
      padding: 10px;
      outline: none;
    }
  }
`;
