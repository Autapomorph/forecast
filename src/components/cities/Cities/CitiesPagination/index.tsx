import React from 'react';
import ReactPaginate from 'react-paginate';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faChevronLeft, faChevronRight } from '@fortawesome/free-solid-svg-icons';

import * as S from './styles';

type Props = {
  total: number;
  perPage: number;
  currentPage: number;
  onPageChange: (page: number) => void;
};

const CitiesPagination = ({
  total,
  perPage,
  currentPage,
  onPageChange,
}: Props): React.ReactElement => {
  const pageCount = Math.ceil(total / perPage);
  const isSinglePage = pageCount === 1;

  const handlePageClick = (selectedItem: { selected: number }): void => {
    onPageChange(selectedItem.selected);
  };

  return (
    <S.CitiesPagination isSinglePage={isSinglePage}>
      <ReactPaginate
        initialPage={currentPage}
        pageCount={pageCount}
        pageRangeDisplayed={0}
        marginPagesDisplayed={0}
        onPageChange={handlePageClick}
        previousLabel={<FontAwesomeIcon icon={faChevronLeft} size="1x" />}
        nextLabel={<FontAwesomeIcon icon={faChevronRight} size="1x" />}
        breakLabel={null}
        containerClassName="pagination"
        pageClassName="page page--hidden page--number page--number--hidden"
        previousClassName="page page--prev"
        nextClassName="page page--next"
        disabledClassName="page page--control page--disabled page--control--disabled"
      />
    </S.CitiesPagination>
  );
};

export default CitiesPagination;
