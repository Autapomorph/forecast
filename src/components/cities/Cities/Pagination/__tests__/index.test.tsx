import React from 'react';
import ReactPaginate from 'react-paginate';
import { shallow } from 'enzyme';

import CitiesPagination from '..';

describe('CitiesPagination tests', () => {
  const props = {
    total: 100,
    perPage: 10,
    currentPage: 1,
    onPageChange: jest.fn(),
  };

  it('renders properly', () => {
    const wrapper = shallow(<CitiesPagination {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call fetchCity on CitiesPaginationTitle click', () => {
    const wrapper = shallow(<CitiesPagination {...props} />);
    // @ts-ignore
    wrapper.find(ReactPaginate).invoke('onPageChange')({ selected: 2 });

    expect(props.onPageChange).toBeCalledWith(2);
  });
});
