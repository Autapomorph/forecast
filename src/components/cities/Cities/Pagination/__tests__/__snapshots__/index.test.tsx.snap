// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitiesPagination tests renders properly 1`] = `
<styles__CitiesPagination
  isSinglePage={false}
>
  <PaginationBoxView
    activeClassName="selected"
    breakLabel={null}
    containerClassName="pagination"
    disableInitialCallback={false}
    disabledClassName="page page--control page--disabled page--control--disabled"
    initialPage={1}
    marginPagesDisplayed={0}
    nextClassName="page page--next"
    nextLabel={
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              320,
              512,
              Array [],
              "f054",
              "M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z",
            ],
            "iconName": "chevron-right",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size="1x"
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    }
    onPageChange={[Function]}
    pageClassName="page page--hidden page--number page--number--hidden"
    pageCount={10}
    pageRangeDisplayed={0}
    previousClassName="page page--prev"
    previousLabel={
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              320,
              512,
              Array [],
              "f053",
              "M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z",
            ],
            "iconName": "chevron-left",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size="1x"
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    }
  />
</styles__CitiesPagination>
`;
